network-api/networkapi/wagtailpages/pagemodels/publications/publication.py

```diff
@@ -116,6 +116,25 @@ def breadcrumb_list(self):
         """
         return Page.objects.ancestor_of(self).type(PublicationPage).live()
 
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+        pages = []
+        for page in self.get_children():
+            if request.user.is_authenticated:
+                # User is logged in, and can preview a page. Get all pages, even drafts.
+                pages.append({
+                    'child': page,
+                    'grandchildren': page.get_children()
+                })
+            elif page.live:
+                # User is not logged in AND this page is live. Only fetch live grandchild pages.
+                pages.append({
+                    'child': page,
+                    'grandchildren': page.get_children().live()
+                })
```

Comment:
```suggestion
            entry = {
                'child': page,
                'grandchildren': page.get_children()
            }

            if not request.user.is_authenticated:
                entry['grandchildren'] = entry['grandchildren'].live()

            pages.append(entry)
```

Reply:
So I tried this last night.. And strangely enough, even though the code is doing the same thing (as far as I can see) it exposes draft pages still. 

Left: Logged in. Right: Incognito (anonymous user)
![image](https://user-images.githubusercontent.com/4743971/101668824-47978e80-3a0e-11eb-9bb9-73e1badeaeed.png)

Anonymous users shouldn't be able to see draft (not-yet published) pages.

Reply:
Ahh I wrote that an immediately saw what I wasnt seeing when I responded. 

`elif page.live:` is the key here.

Reply:
```
            entry = {
                'child': page,
                'grandchildren': page.get_children() if page.live else []
            }
            if not request.user.is_authenticated:
                entry['grandchildren'] = entry['grandchildren'].live()
            pages.append(entry)
```

maybe?